components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/processor/DCRProcessorTest.java

```diff
@@ -48,6 +69,111 @@ public void testGetCallbackPath() throws Exception {
+            assertNotNull(dcrProcessor.process((RegistrationRequest) identityRequest));
+        } else if (request.equals("UnregistrationRequest")) {
+            UnRegistrationHandler unRegistrationHandler = mock(UnRegistrationHandler.class);
+            when(mockHandlerManager.getUnRegistrationHandler(dcrMessageContext)).thenReturn(unRegistrationHandler);
+
+            when(unRegistrationHandler.handle(dcrMessageContext)).thenReturn(new IdentityResponse.IdentityResponseBuilder());
+            assertNotNull(dcrProcessor.process((UnregistrationRequest) identityRequest));
+        }
+    }
+
+    @DataProvider(name = "instanceType&ErrorcodeProvider")
+    public Object[][] getInstanceErrorcode() throws FrameworkClientException {
+
+        RegistrationRequest registrationRequest = mock(RegistrationRequest.class);
+        UnregistrationRequest unregistrationRequest = mock(UnregistrationRequest.class);
+        return new Object[][]{
+                {"RegistrationRequest", registrationRequest, "dummyErrorCode"},
+                {"RegistrationRequest", registrationRequest, ""},
+                {"UnregistrationRequest", unregistrationRequest, "dummyErrorCode"},
+                {"UnregistrationRequest", unregistrationRequest, ""}
+        };
+    }
+
+    @Test(dataProvider = "instanceType&ErrorcodeProvider")
+    public void testProcessWithException(String request, Object identityRequest, String errorCode) throws Exception {
+
+        mockHandlerManager = mock(HandlerManager.class);
+
+        mockStatic(HandlerManager.class);
+        when(HandlerManager.getInstance()).thenReturn(mockHandlerManager);
+
+        DCRMessageContext dcrMessageContext = mock(DCRMessageContext.class);
+        whenNew(DCRMessageContext.class).withArguments(identityRequest).thenReturn(dcrMessageContext);
+
+        if (request.equals("RegistrationRequest")) {
+            RegistrationHandler registrationHandler = mock(RegistrationHandler.class);
+            when(mockHandlerManager.getRegistrationHandler(dcrMessageContext)).thenReturn(registrationHandler);
+
+            if (errorCode.isEmpty()) {
+                doThrow(new DCRException("")).when(registrationHandler).handle(dcrMessageContext);
+            } else {
+                doThrow(new DCRException(errorCode, "")).when(registrationHandler).handle(dcrMessageContext);
+            }
+            try {
+                dcrProcessor.process((RegistrationRequest) identityRequest);
+            } catch (IdentityException ex) {
+                if (errorCode.isEmpty()) {
+                    assertEquals(ex.getErrorCode(), ErrorCodes.BAD_REQUEST.toString());
+                    return;
+                } else {
+                    assertEquals(ex.getErrorCode(), errorCode);
+                    return;
+                }
+            }
+            fail("Expected exception IdentityException not thrown by process method");
```

Comment:
Shall we move this line in to the try block and remove "return" commands within catch block?